test/e2e/scheduling/ubernetes_lite.go

```diff
@@ -52,19 +56,33 @@ var _ = SIGDescribe("Multi-AZ Clusters", func() {
 		msg := fmt.Sprintf("Zone count is %d, only run for multi-zone clusters, skipping test", zoneCount)
 		e2eskipper.SkipUnlessAtLeast(zoneCount, 2, msg)
 		// TODO: SkipUnlessDefaultScheduler() // Non-default schedulers might not spread
+
+		cs = f.ClientSet
+		ns = f.Namespace.Name
+		nodeList = &v1.NodeList{}
+		e2enode.WaitForTotalHealthy(cs, time.Minute)
+		nodeList, err = e2enode.GetReadySchedulableNodes(cs)
+		if err != nil {
+			framework.Logf("Unexpected error occurred: %v", err)
+		}
 	})
 	ginkgo.It("should spread the pods of a service across zones", func() {
-		SpreadServiceOrFail(f, 5*zoneCount, imageutils.GetPauseImageName())
+		SpreadServiceOrFail(f, 5*zoneCount, imageutils.GetPauseImageName(), cs, nodeList, ns)
 	})
 
 	ginkgo.It("should spread the pods of a replication controller across zones", func() {
-		SpreadRCOrFail(f, int32(5*zoneCount), framework.ServeHostnameImage, []string{"serve-hostname"})
+		SpreadRCOrFail(f, int32(5*zoneCount), framework.ServeHostnameImage, []string{"serve-hostname"}, cs, nodeList, ns)
 	})
 })
 
 // SpreadServiceOrFail check that the pods comprising a service
 // get spread evenly across available zones
-func SpreadServiceOrFail(f *framework.Framework, replicaCount int, image string) {
+func SpreadServiceOrFail(f *framework.Framework, replicaCount int, image string, cs clientset.Interface, nodeList *v1.NodeList, ns string) {
+	// make the nodes have balanced cpu,mem usage
+	cleanUp, err := createBalancedPodForNodes(f, cs, ns, nodeList.Items, podRequestedResource, 0.6)
```

Comment:
question: Why cannot we call createBalancedPodForNodes() in ginkgo.BeforeEach() instead?

Reply:
We want to immediately defer the call to `cleanUp` at the end of the individual tests. that won't work if it's called in BeforeEach (cleanUp would just be called at the end of the BeforeEach function)

Reply:
Thanks for your explanation, I got it.
The latest PR seems good for me.
I'd like to see @alculquicondor point.